dafny-lang/dafny
# Let Java properly include runtime sources

### Changes

1. Let `dafny translate java --include-runtime` produce emit the Dafny runtime as .java files, similar to the other back-ends. Previously ` --include-runtime` has no effect when using `dafny translate java`.
1. Let `dafny /compileTarget:java /compile:2 /spillTargetCode:3 /useRuntimeLib` emit sources translated from your Dafny files, a `UserCode.jar` that doesn't include the Dafny runtime, and a `DafnyRuntime.jar`. Previously it would fail on invoking `javac`.
1. Rename RuntimeLib to IncludeRuntimeSources, inverting its meaning

### Testing

- Update the test ManualCompile.dfy so that it no longer uses `DafnyRuntime.jar` from the binary directory, but instead only relies on source files generated by the Dafny binary.
- Add `JavaUseRuntimeLib.dfy` to verify change number 2 works.

Fixes #3611

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>
